### DISCUSSION TITLE
WinMerge License Query

### DISCUSSION THREAD

**Discussion by PallaviVarian** [WinMerge/winmerge#887] [Q&A]
1) If I use WinMerge exe in my application, but don't modify, do I have to release my application under the GPL?

2) What if I modify some WinMerge Source code and use that exe in my application uses. 
   Then do I have to release my application under the GPL, or can I just supply the modified WinMerge code under the GPLs terms.

3) And what if I use WinMerge, but don't modify it, can I distribute it with my application?

**Comment by sdottaka** [comment]
I'm not familiar with licensing, but I'll write an answer. Please point out if it is wrong.

> 1. If I use WinMerge exe in my application, but don't modify, do I have to release my application under the GPL?

If your application is not linked to the WinMerge source code, I don't think you need to release your application under the GPL.

> 2. What if I modify some WinMerge Source code and use that exe in my application uses. Then do I have to release my application under the GPL, or can I just supply the modified WinMerge code under the GPLs terms.

If your application is not linked to the WinMerge source code as above, I don't think you need to release your application under the GPL. However, the person who received both your application and WinMerge must be able to obtain the modified WinMerge source code by some means.

> 3. And what if I use WinMerge, but don't modify it, can I distribute it with my application?

It is possible to distribute your application and WinMerge together. However, the person who received both your application and WinMerge must be able to obtain the WinMerge source code by some means.

See also the FAQ below.

https://www.gnu.org/licenses/old-licenses/gpl-2.0-faq.en.html
